pkg/models/iam/am/am.go

```diff
@@ -19,116 +19,357 @@ package am
+		if contains(roleBinding.Subjects, username) && inSpecifiedWorkspace {
+			result = append(result, roleBinding)
+		}
+	}
+
+	return result, nil
+}
+
+func (am *amOperator) ListClusterRoleBindings(username string) ([]*rbacv1.ClusterRoleBinding, error) {
+
+	roleBindings, err := am.k8sinformer.Rbac().V1().ClusterRoleBindings().Lister().List(labels.Everything())
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]*rbacv1.ClusterRoleBinding, 0)
+
+	for _, roleBinding := range roleBindings {
+		if contains(roleBinding.Subjects, username) {
+			result = append(result, roleBinding)
+		}
+	}
+
+	return result, nil
+}
+
+func (am *amOperator) ListGlobalRoleBindings(username string) ([]*iamv1alpha2.GlobalRoleBinding, error) {
+	roleBindings, err := am.ksinformer.Iam().V1alpha2().GlobalRoleBindings().Lister().List(labels.Everything())
+
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]*iamv1alpha2.GlobalRoleBinding, 0)
+
+	for _, roleBinding := range roleBindings {
+		if contains(roleBinding.Subjects, username) {
+			result = append(result, roleBinding)
+		}
+	}
+
+	return result, nil
+}
+
+func (am *amOperator) ListRoleBindings(username, namespace string) ([]*rbacv1.RoleBinding, error) {
+
+	roleBindings, err := am.k8sinformer.Rbac().V1().RoleBindings().Lister().RoleBindings(namespace).List(labels.Everything())
+
+	if err != nil {
+		klog.Error(err)
+		return nil, err
+	}
+
+	result := make([]*rbacv1.RoleBinding, 0)
+
+	for _, roleBinding := range roleBindings {
+		if contains(roleBinding.Subjects, username) {
+			result = append(result, roleBinding)
+		}
+	}
+
+	return result, nil
+}
+
+func contains(subjects []rbacv1.Subject, username string) bool {
+	for _, subject := range subjects {
+		if subject.Kind == rbacv1.UserKind && (username == "" || subject.Name == username) {
+			return true
+		}
+	}
+	return false
+}
+
+func (am *amOperator) ListRoles(namespace string, query *query.Query) (*api.ListResult, error) {
+	return am.resourceGetter.List("roles", namespace, query)
+}
+
+func (am *amOperator) ListClusterRoles(query *query.Query) (*api.ListResult, error) {
+	return am.resourceGetter.List("clusterroles", "", query)
+}
+
+func (am *amOperator) ListWorkspaceRoles(queryParam *query.Query) (*api.ListResult, error) {
+	return am.resourceGetter.List(iamv1alpha2.ResourcesPluralWorkspaceRole, "", queryParam)
+}
+
+func (am *amOperator) ListGlobalRoles(query *query.Query) (*api.ListResult, error) {
+	return am.resourceGetter.List(iamv1alpha2.ResourcesPluralGlobalRole, "", query)
+}
+
+// GetRoleReferenceRules attempts to resolve the RoleBinding or ClusterRoleBinding.
+func (am *amOperator) GetRoleReferenceRules(roleRef rbacv1.RoleRef, bindingNamespace string) ([]rbacv1.PolicyRule, error) {
+	switch roleRef.Kind {
+	case "Role":
+		role, err := am.k8sinformer.Rbac().V1().Roles().Lister().Roles(bindingNamespace).Get(roleRef.Name)
+		if err != nil {
+			return nil, err
+		}
+		return role.Rules, nil
+
+	case "ClusterRole":
```

Comment:
Hard cord`Role` and `ClusterRole` here is not a good practice

Reply:
done